classes.py

```diff
@@ -0,0 +1,102 @@
+from datetime import datetime
+import pprint
+
+class Menu:
+    
+    def __init__(self, dados):
+        self.dados = dados
+    
+    def mostrar_menu(self):
+        print("==== MENU ====")
+        print("1. Mostrar eventos do presidente.")
+        print("2. Local mais comum dos eventos do presidente.")
+        print("3. Qual é o próximo evento do presidente.")
+        print("4. Print dos eventos por ordem alfabetica do título do evento.")
+        print("5. Print dos eventos por ordem alfabetica do local do evento.")
+        print("6. Print dos eventos ordenados por data (apenas o dia).")
+        print("7. Print dos eventos ordenados por hora (apenas o horario).")
+        print('0. Sair')
+        print("==============")
+        
+    def opcao_1(self):
+        
+        print("\nVocê escolheu a Opção 1!\n")
+        pprint.pprint(f"{self.dados}")
+        
+
+    def opcao_2(self):
+        print("\nVocê escolheu a Opção 2!\n")
+        locais = [evento['local'] for evento in self.dados]
+        local_mais_frequentados = max(set(locais), key=locais.count)
+        print(f"O local mais comum dos eventos é: {local_mais_frequentados}")
+
+    
+    def opcao_3(self):
+        print("\nVocê escolheu a Opção 3!\n")
+        proximos_eventos = []
+        agora = datetime.now()
+
+        eventos_ordenados_data = sorted(self.dados, key=lambda x: (x['dia'], x['horario']))
+        
+        for evento in eventos_ordenados_data:
+            dia_hora = datetime.strptime(f"{evento['dia']} {evento['horario']}", '%Y-%m-%d %Hh%M')
+            if dia_hora > agora:
+                proximos_eventos.append(evento)
```

Comment:
Boa,

mas uma dica: Isso aqui poderia ser alcançado usando a função `filter`.

tipo

```
filter(lambda x: datetime.strptime(f"{evento['dia']} {evento['horario']}", '%Y-%m-%d %Hh%M') > agora, eventos_ordenados_data)
```

mas como essa expressão fica muito longa e difícil para ler, poderiamos simplificar mais as coisas, comparando apenas as strings.

Quando estamos trabalhando com datas no formato `%Y-%m-%d %H:%M` você pode comparar as strings pois é como se as datas pudessem ser organizadas em ordem alfabetica.

então, ao invés de ter um objeto datetime na variável `agora`, a gente pode transformar ela em string para não precisarmos transformar a data do evento em datetime.

Teríamos:

```python
agora = datetime.now().strftime('%Y-%m-%d %H:%M')

eventos_futuros = filter(lambda evento_data: evento_data > agora, eventos_ordenados_data)

if eventos_futuros:
    print(proximos_eventos[0])

```

Reply:
Mas também poderíamos poupar tanto trabalho se ao usar um loop for, a gente consiga parar a execução o quanto antes ao invés de iterar pela lista inteira visto que só queremos o próximo evento.

Visto que já temos uma lista ordenada dos eventos, assim, que acharmos um próximo evento, podemos parar o loop for e já retornar esse próixmo evento.

Implemente isso! 🚀 

Dessa forma, se tivermos uma lista incrivelmente grande, não precisaremos passar por todos os elementos se ela já estiver ordenada.